data/secscan_model/secscan_v4_model.py

```diff
@@ -161,3 +200,40 @@ def register_model_cleanup_callbacks(self, data_model_config):
     @property
     def legacy_api_handler(self):
         raise NotImplementedError("Unsupported for this security scanner version")
+
+
+def features_for(report):
+    """
+    Transforms a Clair v4 `VulnerabilityReport` dict into the standard shape of a 
```

Comment:
So, to ensure this code doesn't explode, please define a JSON schema for the Clair returned document, and validate against it before indexing in. if the validation fails, raise an error to the frontend that we can display.

Reply:
That's a good idea.

Reply:
Implemented [here](https://github.com/quay/quay/pull/259/files#diff-de04d2aed57baa3c456f91913d059b04R183).